### PR TITLE
Issue #8665: Switch expression syntax check update for DefaultComesLast

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8665: Switch expression syntax check update for DefaultComesLast

Diff Regression projects: https://gist.githubusercontent.com/nmancus1/e3988f8092d919b5cbe70f5359c4d9e8/raw/a0f6f5860f8025c19868061dee6e0e40960b992f/projects-to-test-on.properties

Diff Regression config: https://gist.githubusercontent.com/nmancus1/059ef586fdd6e4ab146decd450d5bdc0/raw/5dcd861b1b94b2e92195e6e1b2eb06f62e3bb7f2/DefaultComesLast.txt

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java [hunk 1]
```diff
@@ -176,7 +183,11 @@ public void setSkipIfLastAndSharedWithCase(boolean newValue) {
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
-        if (skipIfLastAndSharedWithCase) {
+
+        // Switch rules are not subject to fall through.
+        final boolean isSwitchRule = defaultGroupAST.getType() == TokenTypes.SWITCH_RULE;
+
+        if (skipIfLastAndSharedWithCase && !isSwitchRule) {
```
Comment (esilkensen) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java:190]: Not related to this PR - the name "skipIfLastAndSharedWithCase" seems a little confusing to me... or, the logic seems to be "skip if (NOT last) and (shared with case)", but I am probably missing something?

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java:190]: This is a public [property](https://checkstyle.sourceforge.io/config_coding.html#DefaultComesLast_Properties), so fixing the name will be a breaking change.

**Comment by pbludov** [comment]
@nmancus1 please fix CI

**Comment by nmancus1** [comment]
> please fix CI

Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java (hunk 1 above)
Comment (esilkensen) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java:190]: I understand, I didn't mean to request it should be changed, just an observation.

**Pull request merged**